scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -760,6 +814,35 @@ private void recover(RecoveredSchedulerState recoveredState) {
         }
     }
 
+    private void recoverTasksState(Vector<InternalJob> jobs, boolean restoreInErrorTasks) {
+        Iterator<InternalJob> iterJob = jobs.iterator();
+        while (iterJob.hasNext()) {
+            InternalJob job = iterJob.next();
+
+            int faultyTasksCount = 0;
+
+            for (InternalTask internalTask : job.getITasks()) {
+                switch (internalTask.getStatus()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1301)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)